sorc/chgres_cube.fd/input_data.F90

```diff
@@ -4956,6 +4958,8 @@ subroutine read_input_sfc_grib2_file(localpet)
      where(slmsk_save == 1) dummy2d_i = 1
    
      call search(dummy2d_8,dummy2d_i,i_input,j_input,1,230)
+   else
```

Comment:
What's this do?

Reply:
This fixes a problem I encountered using HRRR data with surface climo fields. The input data from wgrib2 is 4-bit, but esmf expects 8 bit, so all data has to be converted using the real() command, saving the 8-bit form of the data from dummy2d in to dummy2d_8, which is then broadcast using esmf. Before I added this, if sotyp_from_climo = true, the dummy2d_8 array wouldn't be re-populated with the correct soil (or fill) data from dummy2d, and it would still have the data from the previous use of dummy2d_8, which was skin temperature.

Reply:
Thanks! Good catch!